xml/admin_troubleshooting.xml

```diff
@@ -0,0 +1,318 @@
+    salt-run -l debug state.orchestrate orch.kubernetes > salt-orchestration.logs</command></screen>
+   </step>
+   <step>
+    <para>
+     If the orchestration failed in late state, you can check if
+     <command>etcd</command> and the &kube; cluster is up and running.
+     Log in on a &master_node; and check if <command>etcd</command>
+     cluster is up and running.
+    </para>
+<screen>&prompt.root.master;<command>set -a</command>
+&prompt.root.master;<command>source /etc/sysconfig/etcdctl</command>
+&prompt.root.master;<command>etcdctl member list</command>
+&prompt.root.master;<command>etcdctl endpoint health</command>
+&prompt.root.master;<command>etcdctl endpoint status</command>
+&prompt.root.master;<command>journalctl -u etcd</command></screen>
+    <para>
+     Check if &kube; cluster is up and running.
+    </para>
+<screen>&prompt.root.master;<command>kubectl get nodes</command>
+&prompt.root.master;<command>kubectl cluster-info</command>
+&prompt.root.master;<command>journalctl -u kubelet</command>
+&prompt.root.master;<command>journalctl -u kube-apiserver</command>
+&prompt.root.master;<command>journalctl -u kube-scheduler</command>
+&prompt.root.master;<command>journalctl -u kube-controller-manager</command></screen>
+   </step>
+   <step>
+    <para>
+     Collect all relevant logs with <command>supportconfig</command> on
+     the &master_node; and &admin_node;.
+    </para>
+<screen>&prompt.root.admin;<command>supportconfig</command>
+&prompt.root.admin;<command>tar -xvjf /var/log/nts_*.tbz</command>
+&prompt.root.admin;<command>cd nts*</command>
+&prompt.root.admin;<command>cat etcd.txt kubernetes.txt salt-minion.txt</command></screen>
+<screen>&prompt.root.master;<command>supportconfig</command>
+&prompt.root.master;<command>tar -xvjf /var/log/nts_*.tbz</command>
+&prompt.root.master;<command>cd nts*</command>
+&prompt.root.master;<command>cat etcd.txt kubernetes.txt salt-minion.txt</command></screen>
+   </step>
+  </procedure>
+ </sect1>
+
+ <sect1 xml:id="sec.admin.troubleshooting.failed_update">
+  <title>Recovering from Failed Update</title>
+  <para>
+   See <xref
+   linkend="sec.admin.software.transactional-updates.recovering" />.
+  </para>
+ </sect1>
+
+ <sect1 xml:id="sec.admin.troubleshooting.etcd">
+  <title>Checking <literal>etcd</literal> Health</title>
+  <para>
+   To work with <command>etcdctl</command>, you have to pass parameters
+   with paths to required certificates.
+  </para>
+  <para>
+   Use SSH to log into one of the cluster nodes, for example your first master.
+   The following command provides an example for using <command>etcdctl</command>.
+  </para>
+<screen>&prompt.root;<command>set -a</command>
+&prompt.root;<command>source /etc/sysconfig/etcdctl</command>
+&prompt.root;<command>etcdctl --endpoints $ETCDCTL_ENDPOINT --ca-file $ETCDCTL_CA_FILE \
+    --cert-file $ETCDCTL_CERT_FILE --key-file $ETCDCTL_KEY_FILE cluster-health</command>
```

Comment:
I think this command can be simplified to `etcdctl cluster-health` once that you run the `set -a; source /etc/sysconfig/etcdctl`.